bot/exts/utilities/color.py

```diff
@@ -0,0 +1,229 @@
+import colorsys
+import json
+import pathlib
+import random
+from io import BytesIO
+
+from PIL import Image, ImageColor
+from discord import Embed, File
+from discord.ext import commands
+from discord.ext.commands.errors import BadArgument
+from rapidfuzz import process
+
+from bot.bot import Bot
+from bot.exts.core.extensions import invoke_help_command
+
+THUMBNAIL_SIZE = (80, 80)
+
+
+class Colour(commands.Cog):
+    """Cog for the Colour command."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        with open(pathlib.Path("bot/resources/utilities/ryanzec_colours.json")) as f:
+            self.colour_mapping = json.load(f)
+
+    async def send_colour_response(self, ctx: commands.Context, rgb: tuple[int, int, int]) -> None:
+        """Create and send embed from user given colour information."""
+        r, g, b = rgb
+        name = self._rgb_to_name(rgb)
+        if name == "No match found":
+            name = None
+
+        try:
+            colour_or_color = ctx.invoked_parents[0]
+        except IndexError:
+            colour_or_color = "colour"
```

Comment:
This would never fail, as you need to have `color` or `colour` in `invoked_parents` to run this command right? Or am I missing something?

Reply:
If it gets called without a subcommand I need to handle it (I would get an IndexError).